app/controllers/notification_recipients_controller.rb

```diff
@@ -34,4 +26,27 @@ def find_resource
     super
     @notification_recipient.current_user? || not_found
   end
+
+  def payload
+    Rails.cache.fetch(cache_key, raw: true) do
+      notifications = load_notifications
+
+      {
+        :notifications => notifications,
+        :total => notifications.count
+      }.to_json
+    end
+  end
+
+  def load_notifications
+    NotificationRecipient.
+      where(:user_id => User.current.id, :notification_id => Notification.active).
```

Comment:
I mean this is the current query:
```sql
SELECT "notification_recipients".* 
FROM "notification_recipients" 
WHERE "notification_recipients"."user_id" = $1 
AND "notification_recipients"."notification_id" IN (
    SELECT "notifications"."id" 
    FROM "notifications" 
    WHERE (notifications.expired_at >= '2017-03-20 14:57:26.205932'))  
ORDER BY "notification_recipients"."created_at" ASC
```
but if you anyways load the notifications, it makes sense to do a join and just add another condition and reduce the nested query.